File: codewars/8kyu/array-plus-array/dduck-array-plus-array.py
# https://www.codewars.com/kata/5a2be17aee1aaefe2a000151/
from unittest import TestCase

array_plus_array = lambda arr1, arr2: sum([sum(arr1), sum(arr2)])

TestCase().assertEqual(array_plus_array([1, 2, 3], [4, 5, 6]), 21)
TestCase().assertEqual(array_plus_array([-1, -2, -3], [-4, -5, -6]), -21)
TestCase().assertEqual(array_plus_array([0, 0, 0], [4, 5, 6]), 15)
TestCase().assertEqual(array_plus_array([100, 200, 300], [400, 500, 600]), 2100)

File: codewars/8kyu/array-plus-array/dduck-array-plus-array.js
// https://www.codewars.com/kata/5a2be17aee1aaefe2a000151/

const sum = (arr) => arr.reduce((a, c) => a + c, 0);
const arrayPlusArray = (arr1, arr2) => sum([sum(arr1), sum(arr2)]);

console.log(arrayPlusArray([1, 2, 3], [4, 5, 6]), 21);
console.log(arrayPlusArray([-1, -2, -3], [-4, -5, -6]), -21);
console.log(arrayPlusArray([0, 0, 0], [4, 5, 6]), 15);
console.log(arrayPlusArray([100, 200, 300], [400, 500, 600]), 2100);

